src/utils/showcase.ts

```diff
@@ -0,0 +1,77 @@
+import safeSendMessage from '../utils/safeSendMessage'
+import createProjectEmbed from '../utils/createProjectEmbed'
+
+export interface ShowcaseInput {
+  result: VoteResult
+  isUpvote: boolean
+  isDownvote: boolean
+  isPause: boolean
+  guild: Discord.Guild
+  channel: Discord.TextChannel | Discord.DMChannel | Discord.NewsChannel
+  user: Discord.User
+  reaction: Discord.MessageReaction
+}
+
+export default async function (input: ShowcaseInput): Promise<Discord.Message | undefined> {
+  const { result, guild, channel, user, reaction, isUpvote, isDownvote, isPause } = input
+  const { success, reason, project } = result
+
+  // Alert of errors during the vote process
+  if (!success || !project) {
+    log.error(`Could not register ${user.id}'s vote for project ${project?.name} (ID ${project?.id}): ${reason}`)
+    return await safeSendMessage(channel, '⚠️ Your vote was not possible to register. (Internal error)')
+  }
+
+  // Only one of these can be true per operation depending on what reaction was used
+  const wasApproved = result?.wasApproved === true
+  const wasRejected = result?.wasRejected === true
+  const wasPaused = result?.wasPaused === true
+
+  if (isPause && !isUpvote && !isDownvote) {
+    log.info(`Project ${project.name} (ID ${project.id}) was ${wasPaused ? 'paused' : 'resumed'} by **${user.tag}** (${user.id})`)
+  }
+
+  // If project was approved/rejected, log such and (try to) delete submission post
+  if ((wasApproved || wasRejected) && !wasPaused) {
+    const staffVotes = { up: project.upvotes.staff, down: project.downvotes.staff }
+    const veteranVotes = { up: project.upvotes.veterans, down: project.downvotes.veterans }
+
+    log.info(`Project ${project.name} (ID ${project.id}) was ${wasApproved ? 'approved' : 'rejected'} with ${staffVotes.up + veteranVotes.up} upvotes [Staff/vet spread: ${staffVotes.up} | ${veteranVotes.up}] and ${staffVotes.down + veteranVotes.down} downvotes [Staff/vet spread: ${staffVotes.down} | ${veteranVotes.down}]`)
+
+    const voteSituation = `**Upvotes:** **${project.upvotes.staff}** staff, **${project.upvotes.veterans}** veterans\n**Downvotes:** **${project.downvotes.staff}** staff, **${project.downvotes.veterans}** veterans`
+    await safeSendMessage(channel, `${wasApproved ? '✅' : '❌'} Project **${project.name}** (${project.links.source}, ID ${project.id}) was **${wasApproved ? 'APPROVED' : 'REJECTED'}** by **${user.tag}** (${user.id}) with following vote situation:\n${voteSituation}`)
+
+    try {
+      await reaction.message.delete({ reason: `Project ${wasApproved ? 'approved' : 'rejected'} by ${user.tag} (${user.id})` })
+    } catch (err) {
+      log.error(`Could not delete submission post for ${project.name} (ID ${project.id}): ${err}`)
+      await safeSendMessage(channel, `⚠️ Could not delete project submission post. Please delete message ${project.id} manually. (Discord error)`)
+    }
+  }
+
+  // Post to public showcase
+  if (wasApproved && !wasPaused) {
```

Comment:
See again my previous about only one of these being true at any one time
```suggestion
  if (wasApproved) {
```